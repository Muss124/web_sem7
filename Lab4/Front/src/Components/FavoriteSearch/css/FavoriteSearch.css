.searchform {
    display        : flex;
    justify-content: space-between;
    margin-left    : auto;
    margin-right   : auto;
    width          : 95%;
}

.searchformInside {
    margin-top     : auto;
    margin-bottom  : auto;
    display        : flex;
    justify-content: right;
    margin-right   : 15px;
}

.searchformInside button.btn {
    font-size: 13px;
}

form button.btn {
    font-size: 13px;
}