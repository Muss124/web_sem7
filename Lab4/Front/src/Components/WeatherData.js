import React from "react"
import store from "../store.js"

export class WeatherData extends React.Component {
	render() {
		var WeatherData = this.props.value;
		console.log("here");
		console.log(WeatherData)
		return (
			<div class="WeatherData">
				<div class="WeatherTop">
					<h2>{WeatherData.CityName}</h2>
					<img src={"http://openweathermap.org/img/wn/" + WeatherData.Icon + ".png"} alt=""></img>
					<h1>{WeatherData.Temperature} &#8451;</h1>
					<button type="submit" class="btn btn-secondary" onClick={() => {
						console.log("REMOVE");
						store.dispatch({ type: "FAVORITE_REMOVE", payload: WeatherData.CityName });
					}} >&#10006;</button>
				</div>
				<div class="WeatherMain">
					<div class="WeatherMainEl">
						<div class="WeatherParam">Weather</div>
						<div class="WeatherValue">{WeatherData.Weather}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Humidity (%)</div>
						<div class="WeatherValue">{WeatherData.Humidity}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Pressure (hPa)</div>
						<div class="WeatherValue">{WeatherData.Pressure}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Wind speed (meter/sec)</div>
						<div class="WeatherValue">{WeatherData.Wind}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Longitude </div>
						<div class="WeatherValue">{WeatherData.Longitude}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Latitude</div>
						<div class="WeatherValue">{WeatherData.Latitude}</div>
					</div>
				</div>
			</div>
		);
	}
}