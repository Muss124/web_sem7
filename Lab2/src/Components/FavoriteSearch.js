import React from "react"
import store from "../store.js"
import { itemsFetchData } from "../Actions/reques.js"

export class FavoriteSearch extends React.Component {
    constructor(props) {
        super(props);
        this.state = { value: '' };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }
    handleChange(event) {
        this.setState({ value: event.target.value });
    }

    handleSubmit(event) {
        this.addGeo(this.state.value);
        event.preventDefault();
    }

    addGeo(value) {
        // new shit 
        store.dispatch({ type: "FAVORITE_ADD", payload: { loading: true, city: value, data: [] } });
        store.dispatch(itemsFetchData(value));
        //store.dispatch({ type: "FAVORITE_DATA", payload: value });
        // old shit
        /*
        store.dispatch({ type: "FAVORITE_ADD_BEF", payload: { loading: true, city: value, data: [] } })
        var result = null;

        await fetch('http://api.openweathermap.org/data/2.5/weather?q=' + value + '&APPID=f8e5ebb3f762d1a58aaff7f643d7410b&units=metric')
            .then((res) => res.json()).then((res) => { result = res });
        console.log("recevied");
        console.log(result);
        if (result != null && result["cod"] === 200) {
            store.dispatch({ type: "FAVORITE_ADD_OK", payload: { loading: false, city: value, data: result } })
            //data.push(result);
            //window.localStorage.setItem("favorite", JSON.stringify(data));
            ////this.props.handler();
            ////var data = store.getState();
            ////this.setState({ loading: false, success: true, data: data });
        } else {
            alert("City doesn't founded(")
            store.dispatch({ type: "FAVORITE_ADD_ER", payload: value })
        }
        //window.localStorage.setItem('favorite', JSON.stringify(data));
        */
    }
    render() {
        return (
            <div class="searchform">
                <div>
                    <h1>Favorite</h1>
                </div>
                <div class="searchformInside">
                    <form onSubmit={this.handleSubmit}>
                        <input type="search" value={this.state.value} onChange={this.handleChange}></input>
                        <button type="submit" class="btn btn-secondary btn-circle" >Add</button>
                    </form>
                </div>
            </div>
        );
    }
}