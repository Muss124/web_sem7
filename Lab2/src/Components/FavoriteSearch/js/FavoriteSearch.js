import React from "react"
import store from "../../../store"
import { itemsFetchDataRef, itemsFetchData } from "../../../Actions/reques"
import '../css/FavoriteSearch.css'

export class FavoriteSearch extends React.Component {
    constructor(props) {
        super(props);
        this.state = { value: '' };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }
    handleChange(event) {
        this.setState({ value: event.target.value });
    }

    handleSubmit(event) {
        this.addGeo(this.state.value);
        event.preventDefault();
    }

    addGeo(value) {
        store.dispatch({ type: "FAVORITE_ADD", payload: { city: value, data: [] } });
        store.dispatch(itemsFetchData(value));
    }
    render() {
        return (
            <div className="searchform">
                <div>
                    <h1>Favorite</h1>
                </div>
                <div className="searchformInside">
                    <form onSubmit={this.handleSubmit}>
                        <input type="search" value={this.state.value} onChange={this.handleChange}></input>
                        <button type="submit" className="btn btn-secondary btn-circle" >Add</button>
                    </form>
                    <button className="btn btn-secondary" onClick={() => {
                        console.log("REFRESH");
                        var refData = store.getState();
                        refData.forEach(el => {
                            console.log(el);
                            store.dispatch(itemsFetchDataRef(el.city));
                        });
                    }} >&#8634;</button>
                </div>
            </div>
        );
    }
}