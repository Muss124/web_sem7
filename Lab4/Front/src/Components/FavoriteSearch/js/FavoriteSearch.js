import React from "react"
import store from "../../../store"
import '../css/FavoriteSearch.css'
import FetchCityData from '../../../Actions/FetchCityData'
import RefreshCity from "../../../Actions/RefreshCity"

export class FavoriteSearch extends React.Component {
    constructor(props) {
        super(props);
        this.state = { value: '' };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }
    handleChange(event) {
        this.setState({ value: event.target.value });
    }

    handleSubmit(event) {
        this.addGeo(this.state.value);
        event.preventDefault();
    }

    addGeo(value) {
        console.log("form submited");
        store.dispatch({ type: "FAVORITE_ADD", payload: { CityName : value } });
        store.dispatch(FetchCityData(value));
    }
    render() {
        return (
            <div className="searchform">
                <div>
                    <h1>Favorite</h1>
                </div>
                <div className="searchformInside">
                    <form onSubmit={this.handleSubmit}>
                        <input type="search" value={this.state.value} onChange={this.handleChange}></input>
                        <button type="submit" className="btn btn-secondary btn-circle" >Add</button>
                    </form>
                    <button className="btn btn-secondary" onClick={() => {
                        console.log("REFRESH");
                        var refData = store.getState();
                        refData.forEach(el => {
                            console.log(el);
                            store.dispatch(RefreshCity(el.CityName));
                        });
                    }} >&#8634;</button>
                </div>
            </div>
        );
    }
}