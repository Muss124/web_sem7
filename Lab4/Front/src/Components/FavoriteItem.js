import React from 'react';
import {WeatherData} from "./WeatherData.js"

export class FavoriteItem extends React.Component {
	render() {
		if (typeof this.props.value.loading != "boolean") {
			this.props.value = {
				loading : false, 
				city: this.props.value.CityName,
				data: this.props.value
			}
		}
		if (this.props.value.loading) {
			return (
				<div>
					{this.props.value.city}
					<div class="spinner-border text-danger" role="status">
						<span class="sr-only">Loading...</span>
					</div>
				</div>
			);
		}
		else {
			return (
				<div class="FavoriteItem">
					<WeatherData value={this.props.value.data} />
				</div>
			);
		}
	}
}