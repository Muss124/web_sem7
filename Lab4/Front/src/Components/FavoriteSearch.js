import React from "react"
import store from "../store.js"
import { itemsFetchData } from "../Actions/reques.js"

export class FavoriteSearch extends React.Component {
    constructor(props) {
        super(props);
        this.state = { value: '' };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }
    handleChange(event) {
        this.setState({ value: event.target.value });
    }

    handleSubmit(event) {
        this.addGeo(this.state.value);
        event.preventDefault();
    }

    addGeo(value) {
        // new shit 
        store.dispatch({ type: "FAVORITE_ADD", payload: { loading: true, city: value } });
        store.dispatch(itemsFetchData(value));
    }
    render() {
        return (
            <div class="searchform">
                <div>
                    <h1>Favorite</h1>
                </div>
                <div class="searchformInside">
                    <form onSubmit={this.handleSubmit}>
                        <input type="search" value={this.state.value} onChange={this.handleChange}></input>
                        <button type="submit" class="btn btn-secondary btn-circle" >Add</button>
                    </form>
                </div>
            </div>
        );
    }
}