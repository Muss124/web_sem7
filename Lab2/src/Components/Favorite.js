import React from 'react';
import { FavoriteSearch } from "./FavoriteSearch.js"
import { connect } from 'react-redux'
import { FavoriteItem } from "./FavoriteItem.js"

class Favorite extends React.Component {
    render() {
        return (
            <div>
                <FavoriteSearch />
                {
                    this.props.data.map((favCity, i = 0) => {
                        i++;
                        return ([
                            <div class="FavoriteContainer" key={i}>
                                <FavoriteItem value={favCity}  />
                            </div>
                        ])
                    })
                }

            </div>
        )
    }
}

const mapStateToProps = (state) => {
    console.log(state);
    return {
        data: state
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        setName: () => {
            dispatch({
            });
        }
    };
}
export default connect(mapStateToProps, mapDispatchToProps)(Favorite);