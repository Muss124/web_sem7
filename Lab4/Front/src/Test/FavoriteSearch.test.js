import React from 'react';
import { FavoriteSearch } from '../Components/FavoriteSearch'
import { shallow } from '../enzyme';
import { shallowToJson } from 'enzyme-to-json';


describe('Testing render FavoriteSearch', function () {
    it("Correct render", () => {
        const tree = shallow(<FavoriteSearch />);
        expect(shallowToJson(tree)).toMatchSnapshot();
    });
});
