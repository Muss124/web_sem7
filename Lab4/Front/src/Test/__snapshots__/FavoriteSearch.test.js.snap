// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing render FavoriteSearch Correct render 1`] = `
<div
  className="searchform"
>
  <div>
    <h1>
      Favorite
    </h1>
  </div>
  <div
    className="searchformInside"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="search"
        value=""
      />
      <button
        className="btn btn-secondary btn-circle"
        type="submit"
      >
        Add
      </button>
    </form>
    <button
      className="btn btn-secondary"
      onClick={[Function]}
    >
      â†º
    </button>
  </div>
</div>
`;
